test/gzip.bats

```diff
@@ -0,0 +1,129 @@
+load helpers
+
+function setup() {
+    stacker_setup
+}
+
+function teardown() {
+    cleanup
+}
+
+@test "import tar < 1M" {
+  mkdir -p folder1
+  truncate -s 512k  folder1/file1
+  tar cvf test.tar folder1
+  cat > stacker.yaml <<EOF
+tar:
+  from:
+    type: tar
+    url: test.tar
+EOF
+  stacker build
+  cat oci/index.json | jq .
+  m1=$(cat oci/index.json | jq .manifests[0].digest | sed  's/sha256://' | tr -d \")
+  cat oci/blobs/sha256/"$m1" | jq .
+  l1=$(cat oci/blobs/sha256/"$m1" | jq .layers[0].digest | sed  's/sha256://' | tr -d \")
+  [[ "$($SKOPEO --version)" =~ "skopeo version ${SKOPEO_VERSION}" ]] || {
+    echo "$SKOPEO --version should be ${SKOPEO_VERSION}"
+    exit 1
+  }
+  $SKOPEO copy --format=oci oci:oci:tar containers-storage:test:tar
+  $SKOPEO copy --format=oci containers-storage:test:tar oci:oci:test
+  cat oci/index.json | jq .
+  m2=$(cat oci/index.json | jq .manifests[1].digest | sed  's/sha256://' | tr -d \")
+  cat oci/blobs/sha256/"$m2" | jq .
+  l2=$(cat oci/blobs/sha256/"$m2" | jq .layers[0].digest | sed  's/sha256://' | tr -d \")
+  echo "$l1"
+  echo "$l2"
+  [ "$l1" = "$l2" ]
+}
+
+@test "import tar > 1M" {
+  mkdir -p folder1
+  truncate -s 2m folder1/file1
+  tar cvf test.tar folder1
+  cat > stacker.yaml <<EOF
+tar:
+  from:
+    type: tar
+    url: test.tar
+EOF
+  stacker build
+  cat oci/index.json | jq .
+  m1=$(cat oci/index.json | jq .manifests[0].digest | sed  's/sha256://' | tr -d \")
+  cat oci/blobs/sha256/"$m1" | jq .
+  l1=$(cat oci/blobs/sha256/"$m1" | jq .layers[0].digest | sed  's/sha256://' | tr -d \")
+  [[ "$($SKOPEO --version)" =~ "skopeo version ${SKOPEO_VERSION}" ]] || {
+    echo "$SKOPEO --version should be ${SKOPEO_VERSION}"
+    exit 1
+  }
+  $SKOPEO copy --format=oci oci:oci:tar containers-storage:test:tar
+  $SKOPEO copy --format=oci containers-storage:test:tar oci:oci:test
+  cat oci/index.json | jq .
+  m2=$(cat oci/index.json | jq .manifests[1].digest | sed  's/sha256://' | tr -d \")
+  cat oci/blobs/sha256/"$m2" | jq .
+  l2=$(cat oci/blobs/sha256/"$m2" | jq .layers[0].digest | sed  's/sha256://' | tr -d \")
+  echo "$l1"
+  echo "$l2"
+  [ "$l1" = "$l2" ]
+}
+
+@test "import tar.gz < 1M" {
+  mkdir -p folder1
+  truncate -s 512k folder1/file1
+  tar cvzf test.tar.gz folder1
+  cat > stacker.yaml <<EOF
```

Comment:
Independent tests - so we test for various sizes.

Reply:
and .tar and .tar.gz just to be sure.

Reply:
OK.  I'd try to use a tmpdir so that we know we're not mixing contents between tests.
w.r.t the sizes, there are no issues (AFAIK via my pgzip testing) with files (no tar) smaller than the readbuffer size (1M in our case now).  For tar.gz to be useful (I don't think it is, it's just like tar though less likely to compress any more) your tar.gz needs to be enough bigger than readBuffer size to ensure that we end up reading  two or more blocks (ie, pgzip will actually split up the input and compress in parallel).

Reply:
Fixed.